creusot/src/backend/clone_map/elaborator.rs

```diff
@@ -178,15 +179,40 @@ impl<'tcx> SymbolElaborator<'tcx> {
             names.value(def_id, subst).name
         };
 
-        let sig = named_sig_to_why3(ctx, names, name, &pre_sig, def_id);
+        let mut sig = named_sig_to_why3(ctx, names, name, &pre_sig, def_id);
 
         if CloneLevel::Signature == level_of_item {
             return val(ctx, sig, kind);
         } else if CloneLevel::Contract == level_of_item {
             return val(ctx, sig, kind);
         };
 
-        if item.is_hacked() || ctx.is_logical(def_id) {
+        if let Dependency::Item(did, substs) = item
```

Comment:
you don't need the `if let` here on line 140 we extract the id and substs for you already.